v.divide.training_validation.py

```diff
@@ -45,20 +45,50 @@
 #% answer: 30
 #%end
 
-from grass.script import core as grass
+import grass.script as grass
 import os
 import random
+import atexit
 
+newcol = None
 
-def cleanup(rm_vectors):
+
+def cleanup():
     grass.message(_("Cleaning up..."))
-    nuldev = open(os.devnull, 'w')
-    for rm_v in rm_vectors:
+    if newcol:
+        columns_existing = grass.vector_columns(options['input']).keys()
+        if newcol in columns_existing:
+            grass.run_command(
+                    'v.db.dropcolumn', map=options['input'], columns=newcol)
+
+
+def extract_data(input, output, cats, value):
+    if len(cats) > 20000:
+        newcol = 'train_val_%s' % (os.getpid())
+        columns_existing = grass.vector_columns(options['input']).keys()
+        if newcol not in columns_existing:
+            grass.run_command(
+                'v.db.addcolumn', map=input, columns='%s INTEGER' % (newcol))
+        n = 500
+        for i in range(0, len(cats), n):
+            cats_list = cats[i:i+n]
+            where_list = ["cat='%s'" % (cat) for cat in cats_list]
+            grass.run_command(
+                    'v.db.update', where=' OR '.join(where_list),
```

Comment:
Instead of `OR` you can use something like `where cat in ((,).join(catlist))`, this is much shorter.

Reply:
Thanks!

Reply:
addressed in #3